pkg/controllers/namespace/getter.go

```diff
@@ -129,29 +129,13 @@ type projectGetter struct {
 }
 
 // IsProjectRegistrationNamespace returns whether to watch for ProjectHelmCharts in the provided namespace
-func (g *projectGetter) IsProjectRegistrationNamespace(namespace string) (bool, error) {
-	namespaceObj, err := g.namespaces.Get(namespace, metav1.GetOptions{})
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			// A non-existent namespace is not a project registration namespace
-			return false, nil
-		}
-		return false, err
-	}
```

Comment:
This check is now expected to happen on the caller's end, before they call these functions. This makes it safe to use `IsProjectRegistrationNamespace` in an indexer.